msteams-platform/sbs-botbuilder-conversation-AI.yml

```diff
@@ -0,0 +1,709 @@
+
+    1. Go to Visual Studio Code and open a terminal window.
+    
+    1. Run the following command in the terminal to add the latest version of the Teams AI library library to your application using npm:
+
+       ```bash
+       npm install @microsoft/botbuilder-m365
+
+       ```
+
+    1. Go to the **env** folder and update the Open AI key in the `.env.local.user`file.
+
+       ```
+        SECRET_BOT_PASSWORD=
+        SECRET_OPENAI_API_KEY=sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+       ```
+
+    1. Go to the `config.ts` file and add the following code:
+    
+       ```typescript
+       openAIKey: process.env.SECRET_OPENAI_API_KEY
+       ```
+
+       When you debug the app, the Open AI key defined here process the Open AI key from the `.env.local.user` during local debug.
+
+    1. Go to the `teamsapp.local.yml`file and add the following code after `BOT_ID` and `BOT_PASSWORD`:
+
+       ```teamsapp.local.yml
+       SECRET_OPENAI_API_KEY: ${{SECRET_OPENAI_API_KEY}}
+       ```
+
+       When you run the app locally, the Open AI key defined here is used in the local settings.
+
+    1. Go to the `index.ts` file, and add the following code:
+
+        1. Add the following code to import Teams AI library library:
+
+            ```typescript
+            import { Application, ConversationHistory, DefaultPromptManager, DefaultTurnState, OpenAIModerator, OpenAIPlanner, AI } from '@microsoft/botbuilder-m365';
+            import path from "path";
+            ```
+
+            You need to import the required classes to use the functions and variables available in the class to build your app. You can leverage pre-built templates, and simply add your business logic to this scaffold.
+
+        1. Add the following code to create planner, moderator, and promptManager:
+
+            ```typescript
+            // Create AI components
+            const planner = new OpenAIPlanner({
+            apiKey: config.openAIKey,
+            defaultModel: 'text-davinci-003',
+            logRequests: true
+            });
+            const moderator = new OpenAIModerator({
+            apiKey: config.openAIKey,
+            moderate: 'both'
+            });
+            const promptManager = new DefaultPromptManager(path.join(__dirname, './prompts' ));
+            ````
+            * Planner: The `OpenAIPlanner` is a class that implements an interface called `planner`. The planner calls the LLM model 'text-davinci-003' for text completion. You can use the `OpenAIPlanner` or Azure OpenAI planner. If your using the Azure OpenAI planner, ensure that you define the endpoint property.
+
+            * PromptManager: `promptManager` Allows you to register your prompts, load a prompt template into the AI component. All the functions related to the prompt manager are available in the `DefaultPromptManager`.  The prompt manager parses the prompts defined in the `prompt.txt` folder. You can replace or override your own functionality in the prompt manager. 
```

Comment:
`prompt` folder, not `prompts.txt` folder

Reply:
Updated.